File: testings/equation_solver.c
#include <stdio.h>
#include<math.h>


float function(float x){
    return (float)(pow((double)x,4)-5*pow((double)x,2)+4);
}

float solver(float a,float b,float c){
    float x1=function(a);
    float x2=function(b);

    if(x1==0){
        return a;
    }
    else if(x1*x2<0){
        c=(a+b)/2;
        float xtmp=function(c);
        if(xtmp<0){
            a=c;
        }
        if(xtmp>0){
            b=c;
        }
        if(round(function(c))==0.00){
            return c;
        }
        else{
            solver(a,b,c);
        }

    }
    
}

int main()
{
    for(int i=-10;i<=10;i++){
        if(solver(i,i+1,0))
        printf("%f \n",solver(i,i+1,0));
    }
    return 0;
}

File: testings/equation_solver.py
def function(x):
    #return x**4-3*x**3-10*x**2+24*x
    #return x**3-x**2-12*x
    #return x**2-x-12
    #return x**2+x-6
    #return x**4+2*x**3+11*x**2+10*x+24
    return x**3-x-1


def solver(a,b,c=0):
    print(c)
    x1=function(a)
    x2=function(b)
    if(x1==0):
        print(a)
        return a
    elif(x1*x2<0):
        
        c=(a+b)/2
        xtmp=function(c)
        if(xtmp<0):
            a=c
        elif(xtmp>0):
            b=c
        
        if(round(function(c),2)==0):
            #print(c,"<-This works!")
            return c
        else:
            return solver(a,b,c)
    
lowerRange=int(input("Lower range: "))
upperRange=int(input("Upper range: "))

print("Solution: ")
print(*filter(lambda x: x!=None,[solver(i,i+1) for i in range(lowerRange,upperRange+1)]),"<-But this does not?")
#print([solver(i,i+1) for i in range(lowerRange,upperRange+1)],"<-Why are all the values None?")
        


